src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -2642,8 +2643,326 @@ public override string TypeNameOfValue
+            {
+                case 0: return typeof(PSGenericType0);
+                case 1: return typeof(PSGenericType1);
+                case 2: return typeof(PSGenericType2);
+                case 3: return typeof(PSGenericType3);
+                case 4: return typeof(PSGenericType4);
+                case 5: return typeof(PSGenericType5);
+                case 6: return typeof(PSGenericType6);
+                case 7: return typeof(PSGenericType7);
+                case 8: return typeof(PSGenericType8);
+                case 9: return typeof(PSGenericType9);
+                case 10: return typeof(PSGenericType10);
+                case 11: return typeof(PSGenericType11);
+                case 12: return typeof(PSGenericType12);
+                case 13: return typeof(PSGenericType13);
+                case 14: return typeof(PSGenericType14);
+                case 15: return typeof(PSGenericType15);
+                case 16: return typeof(PSGenericType16);
+            }
+            throw Assert.Unreachable;
+        }
+    }
+
+    class PSGenericType0 : PSGenericType { private PSGenericType0() { } }
+    class PSGenericType1 : PSGenericType { private PSGenericType1() { } }
+    class PSGenericType2 : PSGenericType { private PSGenericType2() { } }
+    class PSGenericType3 : PSGenericType { private PSGenericType3() { } }
+    class PSGenericType4 : PSGenericType { private PSGenericType4() { } }
+    class PSGenericType5 : PSGenericType { private PSGenericType5() { } }
+    class PSGenericType6 : PSGenericType { private PSGenericType6() { } }
+    class PSGenericType7 : PSGenericType { private PSGenericType7() { } }
+    class PSGenericType8 : PSGenericType { private PSGenericType8() { } }
+    class PSGenericType9 : PSGenericType { private PSGenericType9() { } }
+    class PSGenericType10 : PSGenericType { private PSGenericType10() { } }
+    class PSGenericType11 : PSGenericType { private PSGenericType11() { } }
+    class PSGenericType12 : PSGenericType { private PSGenericType12() { } }
+    class PSGenericType13 : PSGenericType { private PSGenericType13() { } }
+    class PSGenericType14 : PSGenericType { private PSGenericType14() { } }
+    class PSGenericType15 : PSGenericType { private PSGenericType15() { } }
+    class PSGenericType16 : PSGenericType { private PSGenericType16() { } }
+
+    internal abstract class MethodGroup { }
+    internal class MethodGroup<T1> : MethodGroup { }
+    internal class MethodGroup<T1, T2> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16> : MethodGroup { }
+    internal class MethodGroup<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16, T17, T18, T19, T20, T21, T22, T23, T24, T25, T26, T27, T28, T29, T30, T31, T32> : MethodGroup { }
+
+    class Unit
```

Comment:
Private class?

Reply:
private is not allowed as a modifier. It is implicit.